components/claim/claim_controller.tsx

```diff
@@ -12,26 +12,33 @@ import EmailToOAuth from 'components/claim/components/email_to_oauth';
 import LDAPToEmail from 'components/claim/components/ldap_to_email';
 import EmailToLDAP from 'components/claim/components/email_to_ldap';
 
-export default class ClaimController extends React.PureComponent {
-    static propTypes = {
-        location: PropTypes.object.isRequired,
-        siteName: PropTypes.string,
-        ldapLoginFieldName: PropTypes.string,
-        passwordConfig: PropTypes.object,
+interface PasswordConfig {
+    minimumLength: number;
+    requireLowercase: boolean;
+    requireUppercase: boolean;
+    requireNumber: boolean;
+    requireSymbol: boolean;
+}
 
-        /*
-         * Object from react-router
-         */
-        match: PropTypes.shape({
-            url: PropTypes.string.isRequired,
-        }).isRequired,
+type Location = {
+    search: string;
+}
 
-        actions: PropTypes.shape({
-            switchLdapToEmail: PropTypes.func.isRequired,
-        }).isRequired,
-    };
+type Props = {
+    location: Location;
+    siteName?: string;
+    ldapLoginFieldName?: string;
+    passwordConfig: PasswordConfig;
```

Comment:
from the previous code, I'm guessing the `passwordconfig` prop is not mandatory
```suggestion
    passwordConfig?: PasswordConfig;
```

Reply:
done